spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/SparkValueConverter.java

```diff
@@ -129,4 +130,44 @@ private static Record convert(Types.StructType struct, Row row) {
     }
     return record;
   }
+
+  public static Object convertToSpark(Type type, Object object) {
+    if (object == null) {
+      return null;
+    }
+
+    switch (type.typeId()) {
+      case STRUCT:
+      case LIST:
+      case MAP:
+        return new UnsupportedOperationException("Complex types currently not supported");
+      case DATE:
+        // if spark.sql.datetime.java8API.enabled is set to true, java.time.LocalDate
```

Comment:
How relevant is this comment?
We needed it for converting to Iceberg as it may be either `Date` or `LocalDate`.
It seems redundant here as we are converting number of days to `LocalDate`.